python/hidet/graph/ops/softmax.py

```diff
@@ -153,3 +156,143 @@ def softmax_kernel(xs: xdtype[shape], ys: xdtype[shape]):
         ir_module = module.ir_module()
 
         return ir_module
+
+    def implement_cpu(self, working_dir: str) -> Union[IRModule, List[IRModule]]:
+        # if not all(is_constant(dim) for dim in self.inputs[0].shape)\
+        if self.inputs[0].type.dtype != float32:
+            return NotImplemented  # use auto-scheduler
+        return tune.extract_ir_modules(self.schedule_softmax_cpu)
+
+    def allow_epilogue(self) -> bool:
+        return False
+
+    def allow_prologue(self) -> bool:
+        return False
```

Comment:
Create a CPU version of the operator because cuda version allows prologue & epilogue.